### ISSUE TITLE
Need to add to Engineer Proceess: Tune Plasma Weapon

### ISSUE THREAD

**Issue by bleehu** [trowl223/Compound_X#323]
We have tune gun and tune laser. We should have tune plasma.

**Comment by Turtlelord26** [comment]
Why are they different abilities in the first place?

Sure the different types of weapon operate differently, but in practice it's all doing the same thing as far as the vast majority of players are concerned. This to me would be an acceptable gamification.

**Comment by bleehu** [comment]
Knowing the details of how 1Source varied plasma, balistic and laser weapons lets us leverage the differences to create mechanically unique bonuses to each. That way we can offer differences in kind and further strategic depth. 
In practice, adding more accuracy to a balistic gun and adding more plasma tokens dealt on hit are fundamentally different.

**Comment by Turtlelord26** [comment]
I like the possibility for difference in kind, but I would like us to consider (if we haven't already) making it one ability with an if-else tree for each of the types of weapon. I suspect it would feel bad as the Engineer to have to allocate three ability slots to be able to buff all of the party's weapons.

**Comment by 1sourcecontrol** [comment]
I tried to list a few main concepts here, for ability design (to be through):
1. Leveraging the difference in kind of the weapon types
2. Difference in magnitude
3. Changing the mechanics of said weapon type
4. Doorway to entry/various tree designs for other engineer abilities

My worst case scenario for this kind of ability design, is for the 3 abilities to be sequential requirements (bullets -> plasma -> lasers on a tree), and in that tree, the abilities improve their respective weapon type with the same magnitude (+15 damage bullets, +2 tokens of plasma burn, -1 miss chance lasers). There is difference in kind, but the same magnitude lessens the value of acquiring further abilities since they will be acquired at sequentially later levels. And even though the abilities affect different stats, they all end up doing one thing, "my weap'n does moar demege". This effectively distills the differences in kind down to "do I want A, B, or C to do more damage?" 

I believe this is where we currently stand:

"Tune laser" is basically a better "Weapon Tuning", but for lasers. 
As seen from the graph,
Weapon Tuning is the bottom node to the branch "material/mechanical" so it's less of a decision for players that want to use that branch. However, Tune Laser is an option to continue higher in another branch. 

I think if we don't give the player different mechanics, it may be advantageous to do what @Turtlelord26 suggested. Consider also having a sort of "weapon tuning I, II, and III". Then you wouldn't lose any space on your trees and the players have a progression that they can still follow if they desire. (you could make it so that the players don't have to get the weapon tuning abilities in any particular order... that way there isn't any pre-reqs). 

Barring that (or in-addition to), my view of the ideal scenario is:

-  If the abilities are acquired at differing levels, they have different magnitude (I bring this up to be thorough; It looks like you are already doing this @bleehu). 

- Abilities grant different mechanics. E.g., an attachment for plasma weapons that fans out/globs the plasma to be like an aoe weapon, shortens it's range, reduces it's accuracy and damage.
A modification to laser weapons that lets them shoot farther at the expense of more ammo. An attachment for guns that lets them switch between different two magazines as a free action or an attachment that just increases damage at the expense of accuracy (pro/con to this are the inherent abundance of bullet-guns will make the bullet version more easily usable).

**Comment by Turtlelord26** [comment]
Obligatory complexity creep warning.

I've been thinking of this as a workhorse ability. If that's correct, then making it too splashy could detract from other abilities that should be splashy. Something has to set the floor by which all the mortars and suicide drones are judged by, and if it isn't a more generic ability of the same class its going to end up being the Soldier. And then Soldiers would be sad.

**Comment by 1sourcecontrol** [comment]
>if it isn't a more generic ability of the same class its going to end up being the Soldier. And then Soldiers would be sad.

@Turtlelord26 I don't understand what you mean, can you please elaborate? 

I don't see a downside to always doing something 'splashy' or cool. Designing for all your abilities to be unique seems like a good goal. Especially if the alternative is trudging through a tree because you have to, in order to get what you want. 
Specifically, in the case that you brought up:
> I suspect it would feel bad as the Engineer to have to allocate three ability slots to be able to buff all of the party's weapons.

If ya'll think it's best to not make these abilities splashy/flashy, then I'd prefer that they function like turtle mentioned, boosting all the weapon types .. and that you can acquire the ability up to 3 times (traversing trees to get there, aka replacing the current Tune weapon abilities) to upgrade its function; or if that seems like too much, removing the other 2 ability spots. 

One more reason I lean towards the 3-types in 1 ability vs 3 abilities that affect effective damage output:
Players are going to encounter fancy weapons of various types. if the engineer has been tuning a marksman's gun, and the marksman gets a cool new plasma long rifle, now the engineer has to invest in a new ability or they aren't part of that fun anymore. 

TLDR; So unless the ability grants a difference in kind (not just increasing effective damage output) or affects all three weapon types, it seems too grindy to me. 

TLDR2; I'm always open to new ideas

**Comment by Turtlelord26** [comment]
Shortish version
There are several aspects at work here, but some of them are
1. There's a limit to how shiny we can make our most basic class. That's hard-coded into reality, especially if we want to be accessible.
2. The more 'unique', 'splashy' effects we have going on, the more complicated the game state becomes. We don't want to make players (or the DM) have to strain too much to keep track of things.
3. (And this is straight from a MtG design column) Relativity is a constant. What I mean by that is that there will always be abilities that are less cool than others, and we neither can nor should avoid that. If every ability was precisely the same amount of interesting, then none of them would be interesting at all. The less interesting is the benchmark by which the humans playing the game will judge the rest of them. *This does not mean we need to deliberately include boring abilities*, only simpler ones. 

Tune Weapon/Armor is exactly the kind of spot where we could do that - make a low-complexity ability that doesn't take a lot of brain processing (which adds up a lot more quickly than you might suspect) to keep track of as the battle goes along, while still providing the Engineer the essential functionality of being a resource generator in any situation.
